integration-tests/contract-multisig-remove-delegate.spec.ts

```diff
@@ -0,0 +1,134 @@
+      expect(delegate).toEqual(null)  
+
+      // Utility function that mimics the PAIR operation of michelson
+      // file deepcode ignore no-any: any is good enough
+      const pair = ({ data, type }: any, value: any) => {
+        return {
+          data: {
+            prim: 'Pair',
+            args: [{ "string": value }, data]
+          },
+          type: {
+            prim: 'pair',
+            args: [{ prim: "address" }, type]
+          }
+        }
+      }
+
+      // Packing the data that need to be sign by each party of the multi-sig
+      // The data passed to this step is specific to this multi-sig implementation
+      // file deepcode ignore no-any: any is good enough
+      const { packed } = await Tezos.rpc.packData(pair({
+        data: {
+          prim: 'Pair',
+          args: [
+            { "int": "0" },
+            {
+              prim: 'Left',
+              args: [MANAGER_LAMBDA.removeDelegate()]
+            }
+          ]
+        } as any,
+        type: {
+          "prim": "pair",
+          "args":
+            [{
+              "prim": "nat",
+              "annots": ["%counter"]
+            },
+            {
+              "prim": "or",
+              "args":
+                [{
+                  "prim": "lambda",
+                  "args":
+                    [{ "prim": "unit" },
+                    {
+                      "prim": "list",
+                      "args":
+                        [{
+                          "prim":
+                            "operation"
+                        }]
+                    }],
+                  "annots":
+                    ["%operation"]
+                },
+                {
+                  "prim": "pair",
+                  "args":
+                    [{
+                      "prim": "nat",
+                      "annots":
+                        ["%threshold"]
+                    },
+                    {
+                      "prim": "list",
+                      "args":
+                        [{ "prim": "key" }],
+                      "annots":
+                        ["%keys"]
+                    }],
+                  "annots":
+                    ["%change_keys"]
+                }],
+              "annots": [":action"]
+            }],
+          "annots": [":payload"]
+        }
+      }, contract.address))
+
+      // Signing the packed
+      const signature1 = await account1.signer.sign(packed, new Uint8Array())
+      const signature2 = await account2.signer.sign(packed, new Uint8Array())
+
+      const op3 = await contract.methods.main(
+        // Counter
+        "0",
+        // Sub function
+        'operation',
+        // Action
+        MANAGER_LAMBDA.removeDelegate(),
+        // Signature list
+        [signature1.prefixSig, signature2.prefixSig, null]
+      ).send()
+
+      await op3.confirmation();
+
+      expect(delegate).toEqual(null)  
```

Comment:
Should a delegate be set first, and then removed? If calling removeDelegate on a contract with no delegate, we might not be testing the correct behavior. 

A suggestion could be to combine the `integration-tests/contract-multisig-set-delegate.spec.ts` and `integration-tests/contract-multisig-remove-delegate.spec.ts` into one test where we 
1. set a delegate 
2. confirm that the delegate is updated
3. remove the delegate
4. confirmed that the delegate is removed

Reply:
i have rewritten the test to work this way and removed the separate removal test.